### PR TITLE
implement and modify internal/io tests

### PR DESCRIPTION
**Pull request by kmrmt**
Signed-off-by: Kosuke Morimoto <kou.morimoto@gmail.com>

### Description:

This PR has implementations that is internal/io package.

### Environment:

- Go Version: 1.17.2
- Docker Version: 20.10.8
- Kubernetes Version: 1.21.5
- NGT Version: 1.13.7

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [X] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your core changes, as applicable?
- [X] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [X] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/io/io_test.go [hunk 1]
```diff
@@ -56,39 +56,53 @@ func TestNewReaderWithContext(t *testing.T) {
...
+			name: "return error with nil context",
+			args: args{
+				ctx: nil,
+				r:   &bytes.Buffer{},
+			},
+			want: want{
+				want: nil,
+				err:  errors.NewErrContextNotProvided(),
+			},
+		},
```
Comment (vankichi) [inline internal/io/io_test.go:98]: could you please use the same table format in 1 test?

File: internal/io/io_test.go [hunk 2]
```diff
@@ -135,39 +149,53 @@ func TestNewReadCloserWithContext(t *testing.T) {
...
+			args: args{
+				ctx: nil,
+				r:   io.NopCloser(&bytes.Buffer{}),
+			},
+			want: want{
+				want: nil,
+				err:  errors.NewErrContextNotProvided(),
+			},
+		},
+	}
```
Comment (vankichi) [inline internal/io/io_test.go:192]: same as above comment

File: internal/io/io_test.go [hunk 3]
```diff
@@ -362,64 +414,77 @@ func TestNewWriterWithContext(t *testing.T) {
...
+			args: args{
+				ctx: nil,
+				w:   &bytes.Buffer{},
+			},
+			want: want{
+				want: nil,
+				err:  errors.NewErrContextNotProvided(),
+			},
+		},
+	}
```
Comment (vankichi) [inline internal/io/io_test.go:481]: same

File: internal/io/io_test.go [hunk 4]
```diff
@@ -467,39 +539,53 @@ func TestNewWriteCloserWithContext(t *testing.T) {
...
+			args: args{
+				ctx: nil,
+				w:   &nopWriteCloser{},
+			},
+			want: want{
+				want: nil,
+				err:  errors.NewErrContextNotProvided(),
+			},
+		},
+	}
```
Comment (vankichi) [inline internal/io/io_test.go:582]: same

**Review by vankichi — COMMENTED** [review]
Almost LGTM
could you refer to [this page](https://vald.vdaas.org/docs/contributing/coding-style/#create-a-table-driven-test) for implementing unit tests and fixing your implementations?

**Comment by kmrmt** [comment]
I fixed the implementation following the testing guidelines.

**Review by vankichi — APPROVED** [review]
LGTM. Thanks :smile:

**Review by kevindiu — APPROVED** [review]
LGTM

**Pull request merged**